### ISSUE TITLE
request-set-header and response-set-header does not behave the same

### ISSUE THREAD

**Issue by cfrantsen** [haproxytech/kubernetes-ingress#345]
The parsing logic for `request-set-header` is limited compared to `response-set-header` resulting in certain header logic not working.

1. Multiline annotations doesn't work unless the last newline is stripped via `|-`. The example in the documentation doesn't work.
```
# Does not work
# ERROR   frontend-annotations.go:409 incorrect value '' in request-set-header annotation
request-set-header: |
    X-Header1 Value1
    X-Header2 Value2

# Works
request-set-header: |-
    X-Header1 Value1
    X-Header2 Value2
```

2. If a header value contains a space an error is thrown, using quotes doesn't matter. The only workaround I have found so far is to use a config-snippet instead.
```
# Does not work
# ERROR   frontend-annotations.go:409 incorrect value 'X-Header1 Value with space' in request-set-header annotation
request-set-header: X-Header1 Value with space
```

A quick look at the codes suggests that reusing the parsing logic from `response-set-header` would fix both of these issues.

**Comment by ivanmatmati** [comment]
Hi @cfrantsen , I've made a fix for request-set-header issues you reported and for response-set-header which is also affected by your 2nd point. It will be available very soon.

**Comment by ivanmatmati** [comment]
The fix has been merged. It should be soon available in nightly build in the commit 40d8125afc3a0af4202171e8a780aaaa47efb953.It will be officially issued in at most one or two weeks.
